lib/matrix.gi

```diff
@@ -168,7 +168,6 @@ InstallOtherMethod( IsDiagonalMat,
     [ IsMatrix ],
     function( mat )
    local  i, j,z;
-    if IsEmpty(mat) then return true;fi;
     z:=Zero(mat[1,1]);
```

Comment:
This, of course, assumes that the all matrix entries have the "same" zero. In some scenarios this may not be the case, though -- at least @frankluebeck mentioned the idea of having, say, a matrix of  polynomials over the rationals, but using integers or rationals as entries instead of the corresponding constant polynomials, to save space and have more efficient arithmetic operations.

In that case, the above optimization won't work correctly.